### PR TITLE
Add login backend logic

### PR DESCRIPTION
**Pull request by patwaririshab**
## Configured Authentication and Set-Up Backend

- Installed required packages Firebase/App, Firebase/FireStore, etc ...
- Used Firebase to set up email authentication (https://rnfirebase.io/auth/usage)
- Added login state management to LoginScreen 
---
#### Authentication Persistence
- In App.js, created an instance of Auth() which checks whether user is already signed in or not:
  - If NOT already signed in:
    - Renders the LoginScreen
  - Else if ALREADY signed in:
    - Renders the navigation.js which loads up the tab based screen

#### Text input state management
- Created state in LoginScreen.js which holds state of username and password
- Upon changeText in either the username or password field, the state is updated

#### Navigation hierarchy
- In NavagationTabInitialiser.js it loads up a bottomTabs instance which has the difference tab screens as its children
- Stored children components (AfterLoginScreen and CameraScreen) in a stack so that they can get automatically linked and I can use backbutton functionality

---
### Testing Details

Steps:
1. Test LoginScreen input fields
  - Expected behaviours: 
    - Username will not auto-capitalise
    - Password will show secureText by default

2. Enter email and password
  - Valid:
    - Expected Behaviour: Loads TabBasedScreen with RecordScreen
  - Invalid:
    - Expected Behaviour: App Crashes with error log

3. Switching between Tab 1, Tab 2, Tab 3
  - Tab 1: RecordScreen
  - Tab 2: React-Native Welcome Screen
  - Tab 3: AfterLoginScreen with 2 buttons (Capture Temp. , View Records)
    - Clicking either buttons will load up CameraScreen with a backButton in header bar
---
### Example of LoginScreen and AfterLoginScreen
|LoginScreen | AfterLoginScreen|
| :--: | :--: |
|Hnet-image (1)|Hnet com-image|

### CODE REVIEW

File: CoronaCare/android/gradle.properties [hunk 1]
```diff
@@ -26,3 +26,7 @@ android.enableJetifier=true
 
 # Version of flipper SDK to use with React Native
 FLIPPER_VERSION=0.33.1
+MYAPP_UPLOAD_STORE_FILE=my-upload-key.keystore
+MYAPP_UPLOAD_KEY_ALIAS=coronacare-uploadkey
+MYAPP_UPLOAD_STORE_PASSWORD=Apple@123
+MYAPP_UPLOAD_KEY_PASSWORD=Apple@123
```
Comment (DarkArcZ) [inline CoronaCare/android/gradle.properties:32]: Why is there a no newline sign beside line 32

File: CoronaCare/src/navigation/NavigationTabInitialiser.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    Navigation.setRoot({
+        root: {
+            bottomTabs: {
+                children: [{
+                    stack: {
+                        children: [{
+                            component: {
+                                name: 'navigation.CoronaCare.RecordsScreen',
+                                passProps: {
+                                    text: 'This is tab 1'
```
Comment (DarkArcZ) [inline CoronaCare/src/navigation/NavigationTabInitialiser.js:14]: why is it so nested

File: CoronaCare/src/screens/AfterLogin.js [hunk 3]
```diff
@@ -9,13 +9,18 @@ export default class Home extends Component {
...
+                name: 'navigation.CoronaCare.CameraScreen'
+              }
         })
+        // Navigation.setRoot({
+        //     root: {
+        //         component: {
+        //             name: "navigation.CoronaCare.CameraScreen"
+        //         }
+        //     }
+        // })
```
Comment (DarkArcZ) [inline CoronaCare/src/screens/AfterLogin.js:23]: nit: Please delete this

File: CoronaCare/src/screens/LoginScreen.js [hunk 4]
```diff
@@ -4,37 +4,73 @@ import { Navigation } from "react-native-navigation"
...
-                    name: "navigation.CoronaCare.WelcomeScreen"
+                    name: "navigation.CoronaCare.NavigationTabInitialiser"
                 }
             }
         });
     }
 
+    sinInUser = () => {
+        console.log(this.state.username)
+        console.log(this.state.password)
```
Comment (DarkArcZ) [inline CoronaCare/src/screens/LoginScreen.js:30]: urgent: Remember to remove

File: CoronaCare/src/screens/LoginScreen.js [hunk 5]
```diff
@@ -71,7 +107,7 @@ const styles = StyleSheet.create({
         height: 40,
         width: 300,
         backgroundColor: 'rgba(255,255,255,0.7)',
-        marginTop: 10, marginBottom: 10,
+        marginTop: 10, marginBottom: 10, paddingLeft: 10,
         borderRadius: 6,
     }
 });
```
Comment (DarkArcZ) [inline CoronaCare/src/screens/LoginScreen.js:113]: Where is there a no newline at the end of line 113

File: CoronaCare/src/navigation/NavigationTabInitialiser.js (hunk 2 above)
Comment (patwaririshab) [reply CoronaCare/src/navigation/NavigationTabInitialiser.js:14]: The Navigation library works in a nested fashion. So a stack has multiple components with the root being the first component to load up. 

In this example: 

`bottomTabs`: loads up a bottomTabs instance and within each tab, it needs to load a screen
`children`:  defines all the child components of the stack (ie. the screens to be shown on each tab)

I am also not very experienced with this, I am pretty sure we can cut down the nesting by reading up on the documentation and figuring out the best way to do it.

**Pull request merged**